lib/puppet/configurer.rb

```diff
@@ -410,8 +410,14 @@ def send_report(report)
   end
 
   def save_last_run_summary(report)
-    mode = Puppet.settings.setting(:lastrunfile).mode
-    Puppet::Util.replace_file(Puppet[:lastrunfile], mode) do |fh|
+    # Warning - this is ugly - if anyone can suggest a better way to do this and get the
+    # corresponding spec tests to work ok - please do.
+    chmode = Puppet.settings.setting(:lastrunfile).mode
+    mode = chmode.to_i(8)
+    # TBD - need this here to check for invalid mode and throw error
+    raise ArgumentError, "#{chmode} is invalid: only octal mode numbers are allowed" unless mode > 0
+
+    Puppet::FileSystem.replace_file(Puppet[:lastrunfile], mode) do |fh|
```

Comment:
Maybe `replace_file` should handle conversion of a string to octal? But this doesn't seem that bad. I'm not sure why we need to store it as a string though.

Reply:
Yeah - I wondered that too but didn't want to change to much on this PR - the string definition has been there for a while.

Reply:
I've revised this slightly as per suggestion by @gimmyxd below.